dspy/teleprompt/bootstrap.py

```diff
@@ -144,36 +136,32 @@ def _bootstrap(self, *, max_bootstraps=None):
 
         bootstrapped = {}
         self.name2traces = {name: [] for name in self.name2predictor}
-
+        progbar = tqdm.tqdm(total=max_bootstraps, desc=f"Bootstrapping {max_bootstraps} examples")
         for round_idx in range(self.max_rounds):
-            for example_idx, example in enumerate(tqdm.tqdm(self.trainset)):
+            if len(bootstrapped) >= max_bootstraps:
+                # If we've already bootstrapped enough examples, don't go to the next round.
+                break
+            for example_idx, example in enumerate(self.trainset):
                 if len(bootstrapped) >= max_bootstraps:
                     break
+                success = self._bootstrap_one_example(example, round_idx)
 
-                if example_idx not in bootstrapped:
-                    success = self._bootstrap_one_example(example, round_idx)
-
-                    if success:
-                        bootstrapped[example_idx] = True
+                if success:
```

Comment:
Did we remove the check below?

```python
if example_idx not in bootstrapped:
```

Reply:
Yea it's my bad, I thought this condition is always true, but in fact the same example can go through bootstrap more than once.